### PR TITLE
feat(issue-153): Open dialog instead of link from pull request

### PR DESCRIPTION
**Pull request by jskurt**
## Issue Link

[issue-153](https://github.com/kafelix496/coding-for-fun/issues/153)

## Notes

-

### CODE REVIEW

File: src/dictionaries/en.ts [hunk 1]
```diff
@@ -18,7 +18,8 @@ export const dictionary = {
     ALERT_DIALOG_DEFAULT_CANCEL_BUTTON: 'Cancel',
     ALERT_DIALOG_DEFAULT_CONTINUE_BUTTON: 'Continue',
     ALERT_DIALOG_DEFAULT_SUBMIT_BUTTON: 'Submit',
-    ALERT_DIALOG_DEFAULT_CONFIRM_BUTTON: 'Confirm'
+    ALERT_DIALOG_DEFAULT_CONFIRM_BUTTON: 'Confirm',
+    ALERT_DIALOG_LINK_TO_PULL_REQUEST_BUTTON: 'Link to Pull Request'
```
Comment (jskurt) [inline src/dictionaries/en.ts:22]: Which is better use 'DEFAULT' or not? 😅
I thought LINK_TO_REQUEST_BUTTON is not default so I didn't use it.

File: src/modules/github/components/pull-list-item/pull-list-item.tsx [hunk 2]
```diff
@@ -90,6 +97,15 @@ export const PullListItem = ({
...
 }: IPullListItemProps) => {
+  const { translate } = useDictionary()
+  const [dialogData, setDialogData] = useState<
+    | {
+        open: true
+      }
+    | {
+        open: false
+      }
+  >({ open: false })
```
Comment (jskurt) [inline src/modules/github/components/pull-list-item/pull-list-item.tsx:108]: Question.
Is it better to use dialog data state for AS? Or is it better to use it as an open state?

File: src/modules/github/components/pull-list-item/pull-list-item.tsx [hunk 3]
```diff
@@ -155,6 +171,20 @@ export const PullListItem = ({
...
 
+  const handleOpenDialog = () => {
+    setDialogData({ open: true })
+  }
+
+  const handleOpenChange = (open: boolean) => {
+    if (!open) {
+      setDialogData({ open })
+    }
+  }
```
Comment (jskurt) [inline src/modules/github/components/pull-list-item/pull-list-item.tsx:182]: Question.
What is purpose that this function?

File: src/modules/github/components/pull-list-item/pull-list-item.tsx [hunk 4]
```diff
@@ -164,13 +209,33 @@ export const PullListItem = ({
...
+              {translate('COMMON.ALERT_DIALOG_DEFAULT_CANCEL_BUTTON')}
+            </Button>
+            <Link href={pull.url} target="_blank">
+              <Button variant="primary">
+                {translate('COMMON.DIALOG_LINK_TO_PULL_REQUEST_BUTTON')}
+              </Button>
+            </Link>
+          </>
+        }
+      ></Dialog>
```
Comment (kafelix496) [inline src/modules/github/components/pull-list-item/pull-list-item.tsx:237]: You don't have to use `<Dialog>` twice in here if there are no children. You can just use it like `<Dialog />`

Comment (jskurt) [reply src/modules/github/components/pull-list-item/pull-list-item.tsx:237]: ahh! okay!

**Review by jskurt — COMMENTED** [review]
abc

**Review by jskurt — COMMENTED** [review]
asdf

File: src/elements/dialog/dialog.tsx [hunk 5]
```diff
@@ -24,7 +24,6 @@ import {
 type TWidthType = 'narrow' | 'default' | 'wide' | 'full'
 
 const DialogRoot = Root
-
```
Comment (kafelix496) [inline src/elements/dialog/dialog.tsx:27]: do not remove enter here.

**Pull request merged**